internal/datastore/spanner/spanner.go

```diff
@@ -221,18 +222,50 @@ func (sd *spannerDatastore) SnapshotReader(revisionRaw datastore.Revision) datas
 	return spannerReader{executor, txSource}
 }
 
+func (sd *spannerDatastore) readTransactionMetadata(ctx context.Context, transactionTag string) (map[string]any, error) {
+	row, err := sd.client.Single().ReadRow(ctx, tableTransactionMetadata, spanner.Key{transactionTag}, []string{colMetadata})
+	if err != nil {
+		if spanner.ErrCode(err) == codes.NotFound {
+			return map[string]any{}, nil
+		}
+
+		return nil, err
+	}
+
+	var metadata map[string]any
+	if err := row.Columns(&metadata); err != nil {
+		return nil, err
+	}
+
+	return metadata, nil
+}
+
 func (sd *spannerDatastore) ReadWriteTx(ctx context.Context, fn datastore.TxUserFunc, opts ...options.RWTOptionsOption) (datastore.Revision, error) {
 	config := options.NewRWTOptionsWithOptions(opts...)
 
 	ctx, span := tracer.Start(ctx, "ReadWriteTx")
 	defer span.End()
 
+	transactionTag := fmt.Sprintf("spicedb-%s", uuid.NewString())
```

Comment:
concatenate instead of `fmt`. Also why prepending `spicedb-`, seems unnecessary waste and UUID should guarantee they are unique, and we already correlate them with the table.

Reply:
Because we might end up using transaction tags of other forms, so I want a prefix